mala/common/parameters.py

```diff
@@ -344,16 +344,25 @@ class ParametersNetwork(ParametersBase):
         network. Please note that the input layer is included therein.
         Default: [10,10,0]
 
-    layer_activations : list
+    layer_activations : list or str
         A list of strings detailing the activation functions to be used
-        by the neural network. If the dimension of layer_activations is
-        smaller than the dimension of layer_sizes-1, than the first entry
-        is used for all layers.
-        Currently supported activation functions are:
-
-            - Sigmoid (default)
-            - ReLU
-            - LeakyReLU
+        by the neural network. If a single string is supplied, then this
+        activation function is used for all layers (including the output layer,
+        i.e., an output activation is used!). Otherwise, the activation
+        functions are added layer by layer.
+        Note that no activation function is applied between input layer and
+        first hidden layer!
+        The items in the list can either be strings, which MALA
+        will map them directly to the correct activation functions or
+        torch.nn.Module objects containing the activation functions directly
+        OR None, in which case no activation function is used. The None
+        can be ommitted at the end, but is useful when layers without
+        activation functions are to be added in the middle
+        Currently supported activation function strings are:
+
+            - "Sigmoid"
+            - "ReLU"
+            - "LeakyReLU" (default)
```

Comment:
`mala.network.network.Network._activation_mappings` also allows "Tanh", so this list needs to be updated. But what is the reasoning behind the restriction to a few functions? Why not

```py
>>> getattr(torch.nn, "ReLU")
torch.nn.modules.activation.ReLU

>>> getattr(torch.nn, "ReLU")()
ReLU()

>>> isinstance(getattr(torch.nn, "ReLU"), torch.nn.Module)
False

>>> isinstance(getattr(torch.nn, "ReLU")(), torch.nn.Module)
True
```

Then `_activation_mappings` could be dropped.

One could also allow people to pass any (list of) `nn.Module` instance(s), like `torch.nn.LeakyReLU(negative_slope=0.02)` instead of strings, which would allow passing arguments and even custom activations, but I'd probably not do this as part of this PR but when the need arises.

Reply:
I modified the internal behavior - now any combination of strings, classes and objects (of course all have to be a subclass of torch.nn.Module) can now be supplied. I also added Tanh to the list.

Reply:
Ah great, thanks re. the option to pass instances (and even classes!).

The other observation I had was that if `activation_function` is a string, then one can always do `getattr(torch.nn, activation_function)()` in `_append_activation_function()`. Then the dict `self._activation_mappings` is not needed and the user is not restricted to the activations in there.

Reply:
Ups, sorry, didn't read your comment carefully enough. I added the `getattr` mechanism. I am using similar mechanisms in other places, so it makes sense to do that here as well. Thanks for the hint!